storagemarket/impl/client.go

```diff
@@ -94,6 +91,7 @@ func NewClient(
 }
 
 func (c *Client) Run(ctx context.Context) {
+	_ = c.restartDeals()
```

Comment:
We should rename this function Start for parity with Provider and also have it return and error so we're not suppressing the error here.